rust/extns/numaflow-sqs/src/source.rs

```diff
@@ -0,0 +1,706 @@
+/// Implementation of the SQS message source using an actor-based architecture.
+///
+/// Key design features:
+/// - Actor model for thread-safe state management
+/// - Batched message handling for efficiency
+/// - Robust error handling and retry logic
+/// - Configurable timeouts and batch sizes
+use std::collections::HashMap;
+use std::time::Duration;
+
+use aws_config::{meta::region::RegionProviderChain, BehaviorVersion};
+use aws_sdk_sqs::config::Region;
+use aws_sdk_sqs::types::{MessageSystemAttributeName, QueueAttributeName};
+use aws_sdk_sqs::Client;
+use bytes::Bytes;
+use chrono::{DateTime, TimeZone, Utc};
+use tokio::sync::{mpsc, oneshot};
+use tokio::time::Instant;
+
+use crate::Error::ActorTaskTerminated;
+use crate::{Error, Result};
+
+/// Configuration for an SQS message source.
+///
+/// Used to initialize the SQS client with region and queue settings.
+/// Implements serde::Deserialize to support loading from configuration files.
+#[derive(serde::Deserialize, Clone, PartialEq)]
+pub struct SQSSourceConfig {
+    pub region: String,
+    pub queue_name: String,
+}
+
+/// Internal message types for the actor implementation.
+///
+/// The actor pattern is used to:
+/// - Ensure thread-safe access to the SQS client
+/// - Manage connection state and retries
+/// - Handle concurrent requests without locks
+enum SQSActorMessage {
+    Receive {
+        respond_to: oneshot::Sender<Result<Vec<SQSMessage>>>,
+        queue_name: String,
```

Comment:
Do we need to have queue_name as part of the message? It will be same for all the messages right?